src/test/python_tests/test_formatting.py

```diff
@@ -138,3 +138,127 @@ def test_skipping_site_packages_files():
 
     expected = None
     assert_that(actual, is_(expected))
+
+
+def test_skipping_excluded_files():
+    """Test skipping formatting when the file is in excluded pattern"""
+
+    UNFORMATTED_EXCLUDED_FILE_PATH = constants.TEST_DATA / "sample3" / "sample.unformatted"
+    with session.LspSession() as ls_session:
+        # Use any stdlib path here
+        uri = utils.as_uri(pathlib.__file__)
+
+        init_options = defaults.VSCODE_DEFAULT_INITIALIZE["initializationOptions"]
+        init_options["settings"][0]["args"] = ["--exclude='**/*.py'"]
+        ls_session.initialize(defaults.VSCODE_DEFAULT_INITIALIZE)
+        ls_session.initialize()
```

Comment:
```suggestion
        init_args = copy.deepcopy(defaults.VSCODE_DEFAULT_INITIALIZE)
        init_options = init_args["initializationOptions"]
        init_options["settings"][0]["args"] = ["--exclude='**/*.py'"]
        ls_session.initialize(init_args)
```

Reply:
`ls_session.initialize()` should only be called once. with the above edit you might also need to import `copy`